src/lib/kdb/kdb_default.c

```diff
@@ -287,7 +287,8 @@ krb5_db_def_fetch_mkey_stash(krb5_context   context,
         goto errout;
     }
 
-    if (!(key->contents = (krb5_octet *)malloc(key->length))) {
+    if (!(key->contents = (krb5_octet *)malloc(
+              sizeof(key->contents[0])*key->length))) {
```

Comment:
That makes more sense.

A bounds check could be good here in principle.  The data is being read from a trusted source, but the file could be corrupt.  The approach we use in cc_file.c is to stat the file and bound all allocations by the file size (itself bounded by SIZE_MAX).  That feels a little cumbersome for a one-off like this, though.  Since keys are relatively small (currently limited to 32 bytes for 256-bit ciphers), we could also just apply a limit like 1024 bytes, which should be good enough for the forseeable future.